typescript/packages/group-tree-plot/src/utils/treePlot.ts

```diff
@@ -0,0 +1,101 @@
+/**
+ * Utilities for handling the tree nodes generated by d3
+ */
+
+import type { D3TreeEdge, D3TreeNode, RecursiveTreeNode } from "../types";
+import type { HierarchyNode } from "d3";
+
+/**
+ * Computes a unique node identifier that should be unique across the entire tree
+ * @param node A node in the tree
+ * @returns A unique identifer
+ */
+export function computeNodeId(node: HierarchyNode<RecursiveTreeNode>): string {
```

Comment:
I wouldn't call this function "compute" as there is nothing really computed. I tend to call these type of functions `makeSomething`, here: `makeNodeId`.